synapse/crypto/keyring.py

```diff
@@ -117,7 +123,29 @@ def callback(ret):
 
     @defer.inlineCallbacks
     def _get_server_verify_key_impl(self, server_name, key_ids):
-        # Try to fetch the key from the remote server.
+        keys = None
+
+        perspective_results = []
+        for perspective_name, perspective_keys in self.perspective_servers.items():
+            @defer.inlineCallbacks
+            def get_key():
+                try:
+                    result = yield self.get_server_verify_key_v2_indirect(
+                        server_name, key_ids, perspective_name, perspective_keys
+                    )
+                    defer.returnValue(result)
+                except:
+                    logging.info(
+                        "Unable to getting key %r for %r from %r",
+                        key_ids, server_name, perspective_name,
+                    )
+            perspective_results.append(get_key())
+
+        perspective_results = yield defer.gatherResults(perspective_results)
```

Comment:
Adding `consumeErrors=True` I *think* ensures that we don't get any `Unhandled Errors`